src/MuonsProducer.cc

```diff
@@ -16,13 +15,22 @@ void MuonsProducer::produce(edm::Event& event, const edm::EventSetup& eventSetup
     const reco::Vertex& primary_vertex = (*vertices_handle)[0];
 
     double rho = *rho_handle;
-
     for (const auto& muon: *muons) {
-        if (! pass_cut(muon))
+	float qter = 1.0;
+        pat::Muon muoncorr=muon;
+        TLorentzVector TLmu;
+        TLmu.SetPxPyPzE(muon.px(),muon.py(),muon.pz(),muon.energy());
+        if(event.isRealData()){
+                rmcor.momcor_data(TLmu, muon.charge(), 0, qter);
+        }
+        else{
+                rmcor.momcor_mc(TLmu, muon.charge(), 0, qter);
+        }
+        math::XYZTLorentzVector lv(TLmu.Px(),TLmu.Py(),TLmu.Pz(),TLmu.E());
+        muoncorr.setP4(lv);
+        if (! pass_cut(muoncorr))
             continue;
-
-        fill_candidate(muon, muon.genParticle());
-
+        fill_candidate(muoncorr, muon.genParticle());
         reco::MuonPFIsolation pfIso = muon.pfIsolationR03();
         computeIsolations_R03(pfIso.sumChargedHadronPt, pfIso.sumNeutralHadronEt, pfIso.sumPhotonEt, pfIso.sumPUPt, muon.pt(), muon.eta(), rho);
```

Comment:
There's an usage of `muon` here which should be `muoncorr`. Please switch all `muon` instances